modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -505,16 +511,20 @@ private void sendMouseEventToFX(MouseEvent e) {
      */
     @Override
     protected void processMouseEvent(MouseEvent e) {
+
+        // Called on EDT thread.
+
+        var hStagePeer = stagePeer;
         if ((e.getID() == MouseEvent.MOUSE_PRESSED) &&
             (e.getButton() == MouseEvent.BUTTON1)) {
             if (isFocusable() && !hasFocus()) {
                 requestFocus();
                 // The extra simulated mouse pressed event is removed by making the JavaFX scene focused.
                 // It is safe, because in JavaFX only the method "setFocused(true)" is called,
                 // which doesn't have any side-effects when called multiple times.
-                if (stagePeer != null) {
+                if (hStagePeer != null) {
                     int focusCause = AbstractEvents.FOCUSEVENT_ACTIVATED;
-                    stagePeer.setFocused(true, focusCause);
+                    hStagePeer.setFocused(true, focusCause);
```

Comment:
suggestion (for a follow up): document which methods in `EmbeddedStageInterface` can be called from which thread.

`setFocused()` method seems to be thread-safe.